jetty-server/src/main/java/org/eclipse/jetty/server/ResourceCache.java

```diff
@@ -246,27 +248,29 @@ private HttpContent load(String pathInContext, Resource resource, int maxBufferS
         {   
             CachedHttpContent content=null;
             
-            // Look for a gzip resource
-            if (_gzip)
+            // Look for a precompressed resources
+            if (_precompressedFormats.length > 0)
             {
-                String pathInContextGz=pathInContext+".gz";
-                CachedHttpContent contentGz = _cache.get(pathInContextGz);
-                if (contentGz==null || !contentGz.isValid())
-                {
-                    contentGz=null;
-                    Resource resourceGz=_factory.getResource(pathInContextGz);
-                    if (resourceGz.exists() && resourceGz.lastModified()>=resource.lastModified() && resourceGz.length()<resource.length())
-                    {
-                        contentGz = new CachedHttpContent(pathInContextGz,resourceGz,null);
-                        CachedHttpContent added = _cache.putIfAbsent(pathInContextGz,contentGz);
-                        if (added!=null)
-                        {
-                            contentGz.invalidate();
-                            contentGz=added;
+                Map<CompressedContentFormat, CachedHttpContent> precompresssedContents=new HashMap<>(_precompressedFormats.length);
```

Comment:
Unfortunately the Trie won't easily work here because the key is of type  CompressedContentFormat (which are constants) instead of String.